spec/lib/workos/audit_logs_spec.rb

```diff
@@ -0,0 +1,145 @@
+# frozen_string_literal: true
+# typed: false
+
+describe WorkOS::AuditLogs do
+  it_behaves_like 'client'
+
+  before do
+    WorkOS.configure do |config|
+      config.key = 'example_api_key'
+    end
+  end
+
+  describe '.create_event' do
+    context 'with valid event payload' do
+      let(:valid_event) do
+        {
+          action: 'user.signed_in',
+          occurred_at: '2022-08-22T15:04:19.704Z',
+          actor: {
+            id: 'user_123',
+            type: 'user',
+            name: 'User',
+            metadata: {
+              foo: 'bar',
+            },
+          },
+          targets: [{
+            id: 'team_123',
+            type: 'team',
+            name: 'Team',
+            metadata: {
+              foo: 'bar',
+            },
+          }],
+          context: {
+            location: '1.1.1.1',
+            user_agent: 'Mozilla',
+          },
+        }
+      end
+
+      context 'with idempotency key' do
+        it 'creates an event' do
+          VCR.use_cassette 'audit_logs/create_event_custom_idempotency_key', match_requests_on: %i[path body headers] do
```

Comment:
On a side note, I noticed that there are many instances of of `VCR.use_cassette` that don't use `match_requests_on` which means it doesn't matter what the mocked requests have defined in their request body. This can lead to very false positive test results.